dev/com.ibm.ws.microprofile.faulttolerance.2.0/src/com/ibm/ws/microprofile/faulttolerance20/state/impl/RetryStateImpl.java

```diff
@@ -111,18 +102,56 @@ private boolean retryOn(Throwable failure) {
     }
 
     private void populateDelay(RetryResultImpl result) {
-        if (jitterStream != null) {
-            result.delay = jitterStream.nextLong();
-        } else {
-            result.delay = delayMillis;
-        }
+        result.delay = delayStream.nextLong();
         if (result.delay < 0) {
             result.delay = 0;
         }
-        result.delayUnit = TimeUnit.MILLISECONDS;
+        result.delayUnit = TimeUnit.NANOSECONDS;
+    }
+
+    /**
+     * Convert a duration to nanoseconds, clamping between MIN_VALUE and MAX_LONG
+     * <p>
+     * Needed in case a user specifies something silly like delay = 5 MILLENNIA
+     *
+     * @param duration the duration to convert
+     * @return duration as nanoseconds, clamped if required
+     */
+    @FFDCIgnore(ArithmeticException.class)
+    private long asClampedNanos(Duration duration) {
```

Comment:
static and pulled into a utils such that it can be unit tested

Reply:
Done